### PR TITLE
Enable identity form on Polkadot

### LINKED ISSUE #6099: enable identity form on Polkadot
I would like to set Identity on Polkadot through KodaDot as we done it on Kusama
- https://wiki.polkadot.network/docs/learn-identity#:~:text=Users%20can%20set%20an%20identity,like%20attestations%20(see%20Judgements).

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #6099 
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?
yes - $

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed UI

<hr/>

### <samp>🤖 Generated by Copilot at 77c1059</samp>

Improved the identity form validation and the profile dropdown navigation. The identity form now handles the submit event properly and checks the input length correctly. The profile dropdown only shows the incoming offers link for chains that support the remark protocol.

### <samp>🤖 Generated by Copilot at 77c1059</samp>

> _To show offers for remarkable art_
> _We check the chain before we start_
> _We use `v-if` and `template`_
> _To render the link as appropriate_
> _And avoid a confusing or empty part_

### CODE REVIEW

File: components/common/IdentityForm.vue [hunk 1]
```diff
@@ -122,13 +124,14 @@ const identity = ref<Record<string, string>>({
   legal: '',
 })
 const deposit = ref('0')
-const inputLengthLimit = ref(32)
+const inputLengthLimit = ref('32')
```
Comment (daiagi) [inline components/common/IdentityForm.vue:127]: to fix a bug where Basic input complain it expects string props

File: components/common/IdentityForm.vue [hunk 2]
```diff
@@ -122,13 +124,14 @@ const identity = ref<Record<string, string>>({
...
 
 onBeforeMount(async () => {
   onApiConnect(apiUrl.value, async (api) => {
     deposit.value = api.consts.identity?.basicDeposit?.toString()
     identity.value = await fetchIdentity(accountId.value)
   })
+  useIdentityStore().fetchBalance({ address: accountId.value })
```
Comment (daiagi) [inline components/common/IdentityForm.vue:134]: fix bug where amount was not always shown in identity form
needed to open wallet modal to make it fetch balance

Comment (roiLeo) [reply components/common/IdentityForm.vue:134]: maybe check getters if empty before invoke `fetchBalance`

File: components/common/IdentityForm.vue (hunk 1 above)
Comment (roiLeo) [reply components/common/IdentityForm.vue:127]: or change `maxlength?: string` to `maxlength?: string | number`

**Review by roiLeo — APPROVED** [review]
✅ code lgtm

bear in mind that https://github.com/kodadot/nft-gallery/issues/6099#issuecomment-1561177316

File: components/common/IdentityForm.vue (hunk 1 above)
Comment (daiagi) [reply components/common/IdentityForm.vue]: done

File: components/common/IdentityForm.vue (hunk 2 above)
Comment (daiagi) [reply components/common/IdentityForm.vue]: done

File: components/navbar/ProfileDropdown.vue [hunk 3]
```diff
@@ -30,12 +30,16 @@
...
             </a>
           </b-dropdown-item>
-          <b-dropdown-item has-link aria-role="menuitem">
-            <nuxt-link :to="{ name: 'identity' }">
-              {{ $t('identity.page') }}
-            </nuxt-link>
-          </b-dropdown-item>
         </template>
+        <b-dropdown-item
+          v-if="chain === 'rmrk' || chain === 'ksm' || chain === 'dot'"
```
Comment (roiLeo) [inline components/navbar/ProfileDropdown.vue:35]: why remove `bsx` and `snek` ? should be available on all chain (maybe not Moonbean)

**Review by Jarsen136 — APPROVED** [review]
wfm

**Comment by prury** [comment]
all chains that support identity seem to be fetching balance properly minus Statemint because i believe it has yet to be implemented. 

Statemine won't show the needed KSM amount for creating an identity tho

Identity created successfully on Basilisk: 
image
image

https://basilisk.subscan.io/extrinsic/3515088-2

@daiagi i believe the charcount is 4px under textfield, alligned right (#6116 )

**Comment by yangwao** [comment]
how u managed to do that? :D 

image

**Comment by yangwao** [comment]
pay 50 usd

**Comment by prury** [comment]
> how u managed to do that? :D
> image

lol, i've deleted a comment of my in this PR because i though it was related to the whole identity form other than only enabling it on Polkadot, but the label can't be removed, so, haha. 

Wasn't able to test this one properly since i'm out of DOT atm

**Pull request merged**